geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnection.java

```diff
@@ -606,38 +605,37 @@ boolean processHandShake() {
         return result;
       }
     } finally {
-      if (isTerminated() || !result) {
-        return false;
-      }
-      boolean registerClient = false;
-      synchronized (getCleanupProxyIdTable()) {
-        MutableInt numRefs = getCleanupProxyIdTable().get(proxyId);
-        if (numRefs != null) {
-          numRefs.increment();
-        } else {
-          registerClient = true;
-          getCleanupProxyIdTable().put(proxyId, new MutableInt(1));
+      if (!isTerminated() && result) {
```

Comment:
It raises a good question. Returning or throwing form a finally block is strongly discouraged. Given the lack of comments it was hard to infer the intent. Given your concern that it may have been intentionally discarding exceptions there is enough reasonable doubt to undo this change. It needs more investigation to understand the intentions.